uefi-services/src/lib.rs

```diff
@@ -104,9 +106,35 @@ fn panic_handler(info: &core::panic::PanicInfo) -> ! {
         }
     }
 
-    loop {
-        // TODO: add a timeout then shutdown.
+    // Give the user some time to read the message
+    if let Some(st) = unsafe { SYSTEM_TABLE } {
+        // FIXME: Check if boot-time services have been exited too
+        st.boot.stall(10_000_000);
+    } else {
+        let mut dummy = 0u64;
+        // FIXME: May need different counter values in debug & release builds
```

Comment:
I tried to look into this, but it seems that the panic hook is optimized out entirely in release builds... :frowning: Guess that's something for another PR.

Reply:
Yeah, it's not a big issue right now. We could probably fix this eventually using some inline assembly and stuff, but it's not a priority.

Reply:
I'm done then :)